Submitted By: DJ Lucas (dj_AT_lucasit_DOT_com)
Date: 2004-05-03
Initial Package Version: 0.77
Origin: Mandrake CVS (Jim Gifford) + cracklib source
Description: Fixes linking on PAM and updates cracklib check

$LastChangedBy$
$Date$


diff -Naur Linux-PAM-0.78-orig/configure.in Linux-PAM-0.78/configure.in
--- Linux-PAM-0.78-orig/configure.in	2004-10-06 09:04:34.000000000 -0500
+++ Linux-PAM-0.78/configure.in	2005-01-30 13:53:16.856763760 -0600
@@ -262,7 +262,7 @@
 
 dnl Look for cracklib dictionary
 AC_MSG_CHECKING(path to cracklib dictionary)
-DICT_DIR_CANDIDATES="/usr/lib /usr/share/dict /usr/share/lib \
+DICT_DIR_CANDIDATES="/lib /usr/lib /usr/share/dict /usr/share/lib \
   /usr/local/lib /usr/local/share/lib"
 DICT_FILE_CANDIDATES="pw_dict cracklib_dict"
 CRACKLIB_DICTPATH=""
@@ -278,7 +278,7 @@
       done
 done
 if test -z "$CRACKLIB_DICTPATH" ; then
-      AC_MSG_ERROR(none found)
+      AC_MSG_RESULT(none found)
 else
       AC_MSG_RESULT($CRACKLIB_DICTPATH)
 fi
diff -Naur Linux-PAM-0.78-orig/libpam_misc/Makefile Linux-PAM-0.78/libpam_misc/Makefile
--- Linux-PAM-0.78-orig/libpam_misc/Makefile	2004-09-28 08:49:19.000000000 -0500
+++ Linux-PAM-0.78/libpam_misc/Makefile	2005-01-30 13:52:46.445386992 -0600
@@ -38,6 +38,8 @@
 SLIBOBJECTS = $(addprefix static/,$(LIBOBJECTS))
 endif
 
+LINKLIBS += -L../libpam -lpam
+
 # ---------------------------------------------
 ## rules
 
diff -Naur Linux-PAM-0.78-orig/modules/pam_cracklib/pam_cracklib.c Linux-PAM-0.78/modules/pam_cracklib/pam_cracklib.c
--- Linux-PAM-0.78-orig/modules/pam_cracklib/pam_cracklib.c	2004-09-15 07:06:17.000000000 -0500
+++ Linux-PAM-0.78/modules/pam_cracklib/pam_cracklib.c	2005-01-30 13:52:46.446386840 -0600
@@ -49,8 +49,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <ctype.h>
-
-extern char *FascistCheck(char *pw, const char *dictpath);
+#include <crack.h>
 
 #ifndef CRACKLIB_DICTPATH
 #define CRACKLIB_DICTPATH "/usr/share/dict/cracklib_dict"
diff -Naur Linux-PAM-0.78-orig/modules/pam_lastlog/Makefile Linux-PAM-0.78/modules/pam_lastlog/Makefile
--- Linux-PAM-0.78-orig/modules/pam_lastlog/Makefile	2001-02-10 16:33:10.000000000 -0600
+++ Linux-PAM-0.78/modules/pam_lastlog/Makefile	2005-01-30 13:52:46.446386840 -0600
@@ -15,5 +15,6 @@
 endif
 
 TITLE=pam_lastlog
+MODULE_SIMPLE_EXTRALIBS=-lutil
 
 include ../Simple.Rules
